File: test/features/step_definitions/pupil-spa/completePageSteps.js
// Step definitions
/* global browser expect */

const {defineSupportCode} = require('cucumber')
defineSupportCode(function ({Given, When, Then}) {
  Given(/^I am on the complete page$/, function () {
    return browser.get(this.config.spaUrl + 'check/complete')
  })

  Then(/^I should see a complete page heading$/, function () {
    return expect(this.spaCompletePage.heading.getText()).to.eventually.equal('Thank you')
  })

  Then(/^I should see some text stating i have completed the check$/, function () {
    return expect(this.spaCompletePage.completionText.getText()).to.eventually.equal('You have completed the multiplication tables check.')
  })

  Then(/^I should be able to sign out$/, function () {
    return expect(this.spaCompletePage.signOut.isPresent()).to.eventually.be.true
  })

  When(/^I choose to sign out$/, function () {
    return this.spaCompletePage.signOut.click()
  })

  Then(/^I should be taken back to the sign in page$/, function () {
    return expect(this.spaSignInPage.schoolPin.isPresent()).to.eventually.be.true
  })
})

File: pupil-spa/test/features/step_definitions/complete_page_steps.rb
Then(/^I should see a complete page heading$/) do
  expect(complete_page).to have_heading
end

Then(/^I should see some text stating i have completed the check$/) do
  sleep(6)
  expect(complete_page).to have_completion_text
end

Then(/^I should be able to sign out$/) do
  expect(complete_page).to have_sign_out
end

When(/^I choose to sign out$/) do
  complete_page.sign_out.click
end

Then(/^I should be taken back to the sign in page$/) do
  expect(sign_in_page).to be_displayed
end
